posthog/temporal/data_imports/pipelines/rest_source/__init__.py

```diff
@@ -244,14 +253,17 @@ def paginate_resource(
                 hooks: Optional[dict[str, Any]],
                 client: RESTClient = client,
                 incremental_object: Optional[Incremental[Any]] = incremental_object,
-                incremental_param: IncrementalParam = incremental_param,
-            ) -> Generator[Any, None, None]:
-                if incremental_object:
+                incremental_param: IncrementalParam | None = incremental_param,
+            ) -> AsyncGenerator[Iterator[Any], Any]:
+                if not is_job_cancelled(team_id=team_id, job_id=job_id):
```

Comment:
It's aborting if the job isn't canceled?

Reply:
Yeah, turns out I was very tired when I was writing this. I also don't even `await` it either 🤡